editor/editor_properties_array_dict.cpp

```diff
@@ -240,6 +234,48 @@ void EditorPropertyArray::_object_id_selected(const StringName &p_property, Obje
 	emit_signal(SNAME("object_id_selected"), p_property, p_id);
 }
 
+void EditorPropertyArray::create_new_property_slot() {
+	int idx = slots.size();
+	HBoxContainer *hbox = memnew(HBoxContainer);
+
+	Button *reorder_button = memnew(Button);
+	reorder_button->set_icon(get_editor_theme_icon(SNAME("TripleBar")));
+	reorder_button->set_default_cursor_shape(Control::CURSOR_MOVE);
+	reorder_button->set_disabled(is_read_only());
+	reorder_button->connect("gui_input", callable_mp(this, &EditorPropertyArray::_reorder_button_gui_input));
+	reorder_button->connect("button_up", callable_mp(this, &EditorPropertyArray::_reorder_button_up));
+	hbox->add_child(reorder_button);
+	EditorProperty *prop = memnew(EditorPropertyNil);
+	hbox->add_child(prop);
+
+	reorder_button->connect("button_down", callable_mp(this, &EditorPropertyArray::_reorder_button_down).bind(idx));
+
+	bool is_untyped_array = object->get_array().get_type() == Variant::ARRAY && subtype == Variant::NIL;
+
+	if (is_untyped_array) {
+		Button *edit_btn = memnew(Button);
+		edit_btn->set_icon(get_editor_theme_icon(SNAME("Edit")));
+		edit_btn->set_disabled(is_read_only());
+		edit_btn->connect("pressed", callable_mp(this, &EditorPropertyArray::_change_type).bind(edit_btn, idx));
+		hbox->add_child(edit_btn);
+	} else {
```

Comment:
This makes it impossible to remove elements in untyped array...

Reply:
This was already like that I just moved this part. I agree that it is not super good but for untyped you need to modify type and there there will be among types a delete item. 
<img width="279" alt="Capture d’écran 2023-09-25 à 10 38 50" src="https://github.com/godotengine/godot/assets/66184050/6b7be0f6-da9e-4738-abdf-1d6c1881a093">